src/datachain/lib/dc/datasets.py

```diff
@@ -147,11 +147,30 @@ def read_dataset(
         namespace_name=namespace,
     )
 
-    if version is not None:
+    # Validate dataset exists and is COMPLETE (filter non-COMPLETE datasets from
```

Comment:
not sure what is happening here .. code and comments look a bit messy . Could you clarify all of this? (and may be cleanup)